libs/ui-lib/src/lib/custom-file-upload/custom-file-upload.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ChangeEvent } from 'react';
+
+import './custom-file-upload.scss';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faImages } from '@fortawesome/free-solid-svg-icons';
+
+export interface CustomFileUploadProps {
+  hasErrors: boolean;
+  errorMessage: string;
+  imageURL: string;
+  handleFileUpload: (event: ChangeEvent<HTMLInputElement>) => void;
+  fieldName: string;
+}
+
+export const CustomFileUpload = ({
+  imageURL,
+  hasErrors,
+  errorMessage,
+  handleFileUpload,
+  fieldName,
+}: CustomFileUploadProps) => {
+  return (
+    <div className="custom-file-upload-container">
+      <label htmlFor={fieldName} className="custom-file-upload">
+        {imageURL ? (
+          <img className="custom-file-upload-image" src={imageURL} alt="Event" />
```

Comment:
@r-saba Brought up a good point that an img in a label doesn't make much sense. Any plans on refactoring this in this ticket? Not having an img in the label might also solve the styling issues I comment on above.

Reply:
Thanks for bringing that up again. I actually think the method @bit172 came up with here is ingenious and easier to read than what I was proposing. The solution I used in another project uses the useRef hook and is not as intuitive to follow. Doing it with pure HTML is much better and since `label` isn't some special element which requires only text, this works perfectly.

The styling issue isn't related to the `label` element.